File: 프로그래머스/unrated/178871. 달리기 경주/달리기 경주.java
import java.util.*;
import java.lang.*;
class Solution {
    public String[] solution(String[] players, String[] callings) {
       Map<String, Integer> playerMap = new HashMap<>();
       Map<Integer,String> rankMap = new HashMap<>();
       for(int i = 0; i < players.length; i++) {
           playerMap.put(players[i],i);
           rankMap.put(i, players[i]);
       }
       for(int i = 0; i < callings.length; i++) {
           int currentRank = playerMap.get(callings[i]);
           int swapedRank = currentRank - 1;
           String swapedPlayer = rankMap.get(swapedRank);
           playerMap.put(callings[i], swapedRank);
           playerMap.put(swapedPlayer, currentRank);
           rankMap.put(swapedRank, callings[i]);
           rankMap.put(currentRank, swapedPlayer);
       }
    return Arrays.stream(rankMap.values().toArray()).toArray(String[]::new);
    }
}

File: 프로그래머스/unrated/178871. 달리기 경주/달리기 경주.js
function solution(players, callings) {
  let playerMap = new Map();
  let rankMap = new Map();
  let result = [];
  for(let i = 0; i < players.length; i++) {
      playerMap.set(players[i],i);
      rankMap.set(i, players[i]);
  }
  for(let i = 0; i < callings.length; i++) {
      let currentRank = playerMap.get(callings[i]);
      let swapedRank = currentRank - 1;
      let swapedPlayer = rankMap.get(currentRank - 1);
      playerMap.set(callings[i],swapedRank);
      playerMap.set(swapedPlayer, currentRank);
      rankMap.set(currentRank, swapedPlayer);
      rankMap.set(swapedRank,callings[i]);
  }

    return Array.from(rankMap.values())
}